doc/services/zbus/index.rst

```diff
@@ -0,0 +1,356 @@
+
+    D: Channel list:
+    D: 0 - Channel acc_data:
+    D:       Message size: 12
+    D:       Observers:
+    D:       - my_listener
+    D:       - my_subscriber
+    D: 1 - Channel version:
+    D:       Message size: 4
+    D:       Observers:
+    D: Observers list:
+    D: 0 - Listener my_listener
+    D: 1 - Subscriber my_subscriber
+
+
+Advanced channel control
+========================
+
+Zbus was designed to be as flexible and extensible as possible. Thus there are some features designed to provide some control and extensibility to the bus.
+
+User Data
+---------
+
+There is a possibility of injecting data into the channel's metadata by enabling the :kconfig:option:`CONFIG_ZBUS_CHANNEL_USER_DATA`. Thus, ``user data`` enables in compile time a memory area of :kconfig:option:`CONFIG_ZBUS_CHANNEL_USER_DATA_SIZE` size into all channels metadata and make this accessible by the ``user_data`` field.
+
+Claim and finish a channel
+--------------------------
+
+To take more control over channels, two function were added :c:func:`zbus_chan_claim` and :c:func:`zbus_chan_finish`. With these functions, it is possible to access the channel's metadata safely. When a channel is claimed, no actions are available to that channel. After finishing the channel, all the actions are available again.
+
+The following code builds on the examples above and claims the ``acc_data`` to set the ``user_data`` to the channel. Suppose we would like to count how many times the channels exchange messages. We defined the ``user_data`` to have the 32 bits integer. This code could be added to the listener code described above.
+
+.. code-block:: c
+
+    if (!zbus_chan_claim(&acc_data, K_NO_WAIT)) {
+            int *message_counting = (int *)acc_data->user_data;
+            *message_counting += 1;
+            zbus_chan_finish(&acc_data);
+    }
+
+Runtime observer registration
+-----------------------------
+
+It is possible to add observers to channels in runtime. This feature uses the freelist technique which the dynamic nodes are pre allocated and can be used and recycled. For enabling that it is necessary to enable the feature :kconfig:option:`CONFIG_ZBUS_RUNTIME_OBSERVERS_REGISTRATION`. It is possible to add or remove a observer to and from a channel. The following example illustrates that.
+
+.. code-block:: c
+
+    ZBUS_LISTENER_DEFINE(my_listener, callback);
+    // ...
+    void thread_entry(void) {
+            // ...
+            /* Adding the observer to channel chan1 */
+            zbus_chan_add_obs(&chan1, &my_listener);
+            /* Removing the observer from channel chan1 */
+            zbus_chan_rm_obs(&chan1, &my_listener);
+
+
+.. note::
+    Zbus can only use a limited number of dynamic observers. This number is 8 by default and can be changed by setting the value of :kconfig:option:`CONFIG_ZBUS_RUNTIME_OBSERVERS_FREELIST_SIZE` as needed. Use the :c:func:`zbus_obs_freelist_count` to verify how many available runtime observers slot exists.
```

Comment:
I would avoid writing down the default config value here (i.e. drop the "This number is 8 by default". It'd be easy to forget to update this if we ever change the default, and the actual value is linked from the kconfig option name anyway.